nixos/tests/ulogd.nix

```diff
@@ -0,0 +1,84 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "ulogd";
+
+  meta = with lib; {
+    maintainers = with maintainers; [ p-h ];
+  };
+
+  nodes.machine = { ... }: {
+    networking.firewall.enable = false;
+    networking.nftables.enable = true;
+    networking.nftables.ruleset = ''
+      table inet filter {
+        chain input {
+          type filter hook input priority 0;
+          log group 2 accept
+        }
+
+        chain output {
+          type filter hook output priority 0; policy accept;
+          log group 2 accept
+        }
+
+        chain forward {
+          type filter hook forward priority 0; policy drop;
+          log group 2 accept
+        }
+
+      }
+    '';
+    services.ulogd = {
+      enable = true;
+      settings = {
+        global = {
+          logfile = "/var/log/ulogd.log";
+          stack = "log1:NFLOG,base1:BASE,pcap1:PCAP";
+        };
+
+        log1.group = 2;
+
+        pcap1 = {
+          file = "/var/log/ulogd.pcap";
+          sync = 1;
+        };
+      };
+    };
+
+    environment.systemPackages = with pkgs; [
+      tcpdump
+    ];
+  };
+
+  testScript = ''
+    start_all()
+    machine.wait_for_unit("ulogd.service")
+    machine.wait_for_unit("network-online.target")
+
+    with subtest("Ulogd is running"):
+        machine.succeed("pgrep ulogd >&2")
+
+    # All packets show up twice in the logs
+    with subtest("Logs are collected"):
+        machine.succeed("ping -f 127.0.0.1 -c 5 >&2")
+        machine.succeed("sleep 2")
```

Comment:
How is this sleep going to fare under a heavy load machine such as the Hydra builders?
Do we have any more reliable to wait for the ulogd to be filled with some timeout?

Reply:
> potential flakiness of the test due to sleep
Maybe stopping the daemon after the test before reading might work

> licenses stuff

linuxdoc-tools is also licensed partially under the sgmls licenseand the iso-entities license. I don't know how else to model this.

Reply:
Stopping doesn't work either. Some packages seem to be missed then as well.

Reply:
`[ sgmls-license iso-entities-license ]` (replacing with their proper standard identifiers) is fine IMHO

Reply:
Ah, I now see that iso-entities and entity-map are just licensed under MIT. I cannot find out what the SGML parser is licensed under. It seems to be its own license and [gentoo seems to treat it as its own license too](https://gitweb.gentoo.org/repo/gentoo.git/tree/licenses/SGMLUG?id=7d999af4a47bf55e53e54713d98d145f935935c1)

Reply:
Awesome!